borsh-derive-internal/src/attribute_helpers.rs

```diff
@@ -42,6 +46,81 @@ pub(crate) fn contains_skip(attrs: &[Attribute]) -> bool {
     attrs.iter().any(|attr| attr.path() == SKIP)
 }
 
+pub fn check_item_attributes(derive_input: &DeriveInput) -> Result<(), TokenStream> {
+    for attr in &derive_input.attrs {
+        if attr.path().is_ident(BORSH.0) {
+            attr.parse_nested_meta(|meta| {
+                if !meta.path.is_ident(USE_DISCRIMINANT) {
+                    return Err(syn::Error::new(
+                        derive_input.ident.span(),
+                        "`use_discriminant` is the only supported attribute for `borsh`",
+                    ));
+                }
+                if meta.path.is_ident(USE_DISCRIMINANT) {
+                    let _expr: Expr = meta.value()?.parse()?;
+                    if let syn::Data::Struct(ref _data) = derive_input.data {
+                        return Err(syn::Error::new(
+                            derive_input.ident.span(),
+                            "borsh(use_discriminant=<bool>) does not support structs",
+                        ));
+                    }
+                }
+
+                Ok(())
+            })
+            .map_err(|err| err.to_compile_error())?;
+        }
+    }
+    Ok(())
+}
+
+pub(crate) fn contains_use_discriminant(input: &ItemEnum) -> Result<bool, syn::Error> {
+    if input.variants.len() >= 256 {
```

Comment:
```suggestion
    if input.variants.len() > 256 {
```
u8 represents 256 variants just fine, 
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=bb3687db94a0b04caab03ece90ebfb6d